File: src/ice/world/blocks/environment/IceFloor.java
package ice.world.blocks.environment;

import arc.Core;
import arc.graphics.g2d.TextureRegion;
import arc.struct.ObjectMap;
import mindustry.world.blocks.environment.Floor;

public class IceFloor extends Floor {
    public static ObjectMap<String, Floor> floors = new ObjectMap<>();

    public IceFloor(String name, int variants) {
        super(name, variants);
        floors.put(name, this);
    }

    @Override
    public TextureRegion[] icons() {
        if (variants==0)return new TextureRegion[]{Core.atlas.find(name)};
        return new TextureRegion[]{Core.atlas.find(name + "1")};
    }
}

File: src/ice/library/type/baseContent/blocks/environment/IceFloor.kt
package ice.library.type.baseContent.blocks.environment

import arc.Core
import arc.struct.Seq
import ice.library.util.accessField
import mindustry.Vars
import mindustry.graphics.BlockRenderer
import mindustry.world.Tile
import mindustry.world.blocks.environment.Floor

open class IceFloor(name: String, variants: Int) : Floor(name, variants) {
    var BlockRenderer.updateFloors: Seq<UpdateRenderState> by accessField("updateFloors")
    var updateFloor = false

    init {
        hasColor = true
    }

    override fun init() {
        super.init()
        Vars.content.block("${this.name}Wall")?.let {
            wall = it
        }
    }

    override fun updateRender(tile: Tile) = updateFloor
    override fun icons() = if (variants == 0) arrayOf(Core.atlas.find(name)) else arrayOf(Core.atlas.find(name + "1"))

}